src/group.rs

```diff
@@ -97,12 +100,11 @@ impl Group for RistrettoPoint {
     }
 
     type UniformBytesLen = U64;
-    fn hash_to_curve(uniform_bytes: &GenericArray<u8, Self::UniformBytesLen>) -> Self {
-        // This is because RistrettoPoint is on an obsolete sha2 version, see https://github.com/dalek-cryptography/curve25519-dalek/pull/327
+    fn hash_to_curve<H: Hash>(
+        uniform_bytes: &GenericArray<u8, Self::UniformBytesLen>,
+    ) -> Self {
         let mut bits = [0u8; 64];
-        let mut hasher = sha2::Sha512::new();
-        hasher.update(uniform_bytes);
-        bits.copy_from_slice(&hasher.finalize());
```

Comment:
Aren't we missing the actual hashing here? The point of the above three lines was to deterministically reproduce the results of [`Scalar::hash_from_bytes`](https://github.com/dalek-cryptography/curve25519-dalek/blob/c4824e1384ef188eb9284ff866e5ad6b814400f7/src/scalar.rs#L587-L593)

Reply:
Discussed offline -- this is not necessary since we are already hashing before passing into this function